File: backend/app/schemas/notification.py
import uuid
from datetime import datetime

from app.models.notification import NotificationType
from app.schemas.base import CamelModel


class NotificationOut(CamelModel):
    id: uuid.UUID
    type: NotificationType
    message: str
    appointment_id: uuid.UUID | None = None
    is_read: bool
    created_at: datetime

File: src/lib/types/notification.ts
export type NotificationType =
  | "APPOINTMENT_CONFIRMED"
  | "APPOINTMENT_CANCELLED"
  | "APPOINTMENT_RESCHEDULED";

export interface AppNotification {
  id: string;
  type: NotificationType;
  message: string;
  appointmentId?: string | null;
  isRead: boolean;
  createdAt: string;
}
